File: src/main/kotlin/sample/ManageUsers.kt
package sample

import sample.proto.UserProfile
import com.google.protobuf.ByteString

fun createKotlinUser(
    userId: String,
    displayName: String,
    age: Int,
    isActive: Boolean,
    interests: List<String>
): UserProfile {
    return UserProfile.newBuilder()
        .setUserId(userId)
        .setDisplayName(displayName)
        .setAge(age)
        .setIsActive(isActive)
        .addAllInterests(interests) 
        .build() 
}

File: src/main/python/sample/manage_users.py
from src.main.proto.sample import user_profile_pb2


def create_new_user(user_id: str, name: str, age: int, is_active: bool, interests: list[str]) -> user_profile_pb2.UserProfile:
    profile = user_profile_pb2.UserProfile()
    profile.user_id = user_id
    profile.display_name = name
    profile.age = age
    profile.is_active = is_active
    profile.interests.extend(interests)
    return profile
